app/code/Magento/Backend/Api/Search/ItemsInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ *
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Backend\Api\Search;
+
+/**
+ * @api
+ */
+interface ItemsInterface
+{
+    const LIMIT = 'limit';
+    const START = 'start';
+    const QUERY = 'query';
+
+    /**
+     * get the search result items
+     *
+     * @return array
+     */
+    public function getResults();
```

Comment:
This interface is pretty bad one because it has a [temporal coupling ](http://blog.ploeh.dk/2011/05/24/DesignSmellTemporalCoupling/)
on the level of contract. Because contract implies an order in which these methods would be called.
Firstly initialize all setters, and then call getResuls()

BTW This interface looks like a good application of Builder pattern for me. 
So, if we would rename it to SearchResultBuilder that will make more sense.